### DISCUSSION TITLE
Convert object column to string

### DISCUSSION THREAD

**Discussion by romikforest** [vaexio/vaex#1667] [Q&A]
Hi. I read some excel file with pandas and convert it to vaex after that. That file have columns supposed to be strings but the text some times contains numbers inside. So as a result I have a column of object type with some elements of type strings and other int64. I tried to convert it to strings, but could do it only using apply. Because e.g. `.astype(str)`, `.to_string()` don't work (TypeError: to_string(): incompatible function arguments. The following argument types are supported...). Is there a simple and fast way to convert object column to strings?

```
f = pd.read_excel(xlsx_path, engine='openpyxl')
f = vx.from_pandas(f)
# f['obj_column'] = f.apply(str, [f['obj_column']])
```

I also managed to read the file correctly using `dtype` argument of `read_excel`. But I've spent some time experimenting with object column conversion and it interests me.

{'vaex-core': '4.5.1', 'vaex-hdf5': '0.10.0'}

**Comment by JovanVeljanoski** [comment]
Hi, 

Can you tell us which version of vaex are you using? 
I was under the impression that when a column have mixed type from reading csv, vaex tries to find "the common denominator" and if everything fails, everything is cast to string.. 

Can you perhaps include a brief code example?

**Reply by romikforest** [reply]
Hi. Sorry I was not accurate. I get that column converting pandas dataframe. I read excel file with pandas first because I didn't find a method to read excel file in vaex. I've updated the question.

**Reply by romikforest** [reply]
I also added version information {'vaex-core': '4.5.1', 'vaex-hdf5': '0.10.0'}

**Reply by JovanVeljanoski** [reply]
Basically vaex does not support "objects". I think short term your best bet is to handle this on the pandas side before passing the data to vaex. 

@maartenbreddels when reading the csv the problem in the OP does not happen (mixed types are cast to string). But when just using `from_pandas`, we are stuck with objects. But I see in the code that `_from_csv_read(...)` simply uses pandas to read the data and then `from_pandas` to create a vaex dataframe.. so Is this a bug somehow? (i can't see it atm).

**Reply by romikforest** [reply]
I used your code for csv as a template to write my own function from_excel.
<img width="999" alt="изображение" src="https://user-images.githubusercontent.com/14263879/138616696-41231ffc-2a4d-41fa-b3fa-822b93b2a01b.png">
I just noticed fs_options has default as dict, that will be initialized during function declaration and not a call. Possible It may be a problem in some cases.
